### ISSUE TITLE
Does not work with Tests that use ES6 Imports

### ISSUE THREAD

**Issue by JKomoroski** [pausan/mocha-list-tests#3]
When running the list test file against test files that use ES6 style imports I get the attached error. If I rename the files to use the `.mjs` the files don't seem to be scanned either. I wasn't able to implement a fix for the current implementation seems to hinge on the use of `require()` inside a loop.

Observed error:
```
Fatal Error (try --help for help):
/My/File/here.js:1
import { A, B } from '../../../x/y';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:1018:16)
    at Module._compile (node:internal/modules/cjs/loader:1066:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1131:10)
    at Module.load (node:internal/modules/cjs/loader:967:32)
    at Function.Module._load (node:internal/modules/cjs/loader:807:14)
    at Module.require (node:internal/modules/cjs/loader:991:19)
    at require (node:internal/modules/cjs/helpers:92:18)
    at findSuitesAndTests (/MY/REPO/node_modules/mocha-list-tests/mocha-list-tests.js:191:7)
    at main (/MY/REPO/node_modules/mocha-list-tests/mocha-list-tests.js:219:18)
    at Object.<anonymous> (/MY/REPO/node_modules/mocha-list-tests/mocha-list-tests.js:230:19)
```

**Comment by pausan** [comment]
I've created a test and added support for ES6 modules while keeping CommonJS modules. Master branch and npm module (https://www.npmjs.com/package/mocha-list-tests v1.0.4) are both updated 

This error happened because ES6 uses "import" instead of "require" statements to load stuff, but while require is a synchronous operation, import is asynchronous, thus, I had to rework the library a little bit to be used from the commandline.

ES6 style modules need to use .mjs extension to be properly recognized by the library, otherwise the library will use require instead.

BTW Today I've also merged some changes from another fork of the project, which adds some extra support and returns more information on each test case by referencing the file/line in which they are defined.

**Comment by JKomoroski** [comment]
I updated to version 1.0.4, renamed all my test files to the `.mjs` extension, and ran the command as before. I'm no longer getting an error, but the json output is empty:
```
{
  "suites": [],
  "tests": [],
  "tree": {}
}
```

I'll have to do some more work to see what is going wrong for me.

**Comment by pausan** [comment]
Can you provide a short/minimal example to try to reproduce it locally?

Please include:
- Node version you are using
- Which command are you using
- Folder structure for the minimal examples
- Code of the minimal example

If you could do an example with 2 or 3 files at most (let's say less than a handful of lines each) that would be awesome. 

Thanks

**Comment by JKomoroski** [comment]
This is a cypress project which uses mocha. Here are some details:

Node Version:
```
node --version
v15.2.0
```
Command:
```node ./node_modules/mocha-list-tests/mocha-list-tests.js my_project/integration```

Folder Structure:
```
tree my_project/integration
.
└── administration
    ├── bugs
    │   └── UI-2143.bug.mjs
    └── regression
        └── admin-next.regression.mjs

3 directories, 2 files
```
Example Files:

UI-2143.bug.mjs
```import { BE_HIDDEN, EXIST } from '../../../support/assertions';

describe("Bug for UI-2143 hasn't returned...", () => {
  const partitionDropdown =
    '.partition-select > .button-mode > [data-cy=select-select-container] > [data-cy=select-button] > [data-cy=button-label] > [data-cy=label-content-container] > :nth-child(1) > [data-cy=button-annotation] > [data-cy=annotation-annotation-validation-container] > [data-cy=annotation-content-container] > [data-cy=button-tooltip-container] > [data-cy=button-tooltip] > [data-cy=tooltip-element-target] > [data-cy=button-button]';
  const dropdownPanel =
    '.partition-select > alz-select.ng-star-inserted > .select-container > alz-panel-view > .panel > .panel-scrolling-container > .panel-content-container';

  before(() => {
    cy.login('alzAdmin');
    cy.forceVisitHarness('/select-input');
  });

  it('UI-2143 - dropdown panel closes after select element is selected', () => {
    cy.showBugLink(__filename);
    cy.get(partitionDropdown).click();
    cy.get(dropdownPanel).should(EXIST);
    cy.get(partitionDropdown).click();
    cy.get(dropdownPanel).should(BE_HIDDEN);
  });
});
```

admin-next.regression.mjs
```
import { header } from '../../../support/page-objects/shared-components.object';
import { HAVE_CLASS } from '../../../support/assertions';

describe('Testing regressions...', () => {
  beforeEach(() => {
    cy.login('alzAdmin');
    cy.createAdvancedAccount();
  });

  it('Initial tab is determined by current URL', () => {
    cy.visit('/');
    cy.url().should('include', 'accounts/recent');
    cy.forceVisit('alianza/partitions');
    cy.get(header.ALIANZA_TAB).should(HAVE_CLASS, 'active');
    cy.get(header.ALIANZA_TAB).should(HAVE_CLASS, 'enabled');

    cy.forceVisit('partitions/1/management/service-activation');
    cy.url().should('include', 'management/service-activation');
    cy.get(header.MANAGEMENT_TAB).should(HAVE_CLASS, 'active');
    cy.get(header.MANAGEMENT_TAB).should(HAVE_CLASS, 'enabled');

    cy.visit('/');
    cy.get(header.SEARCH_INPUT).click().type(Cypress.env('accountNumber'));
    cy.get(header.DROPDOWN_RESULTS).contains(Cypress.env('accountNumber')).click();
    cy.get(header.ACCOUNT_TAB).should(HAVE_CLASS, 'active');
    cy.get(header.ACCOUNT_TAB).should(HAVE_CLASS, 'enabled');
  });

  afterEach(() => {
    cy.deleteAccount();
  });
});
```

**Comment by pausan** [comment]
Bug found! There was a problem when trying to find all files inside a folder with extension mjs. It should find and list all tests now (published version 1.0.5)

**Comment by JKomoroski** [comment]
I was able to test this today and confirmed that it was working. 

I had some import issues where I had to rename some imported `.js` files to `.mjs`. I then had to explicitly use that extension in the import statements across my test classes.

Not sure exactly why, but it's working. Thanks!
